tests/checks/test-BatcacheVariantCheck.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+require_once( 'CodeCheckTestBase.php' );
+
+class BatcacheVariantTest extends CodeCheckTestBase {
+
+	public function testBatcacheVariant() {
+		$expected_errors = array(
+			array(
+				'slug' => 'batcache-variant-error',
+				'level' => BaseScanner::LEVEL_BLOCKER,
+				'description' => 'Illegal word in variant determiner.',
+				'file' => 'BatcacheVariantTest.inc',
+				'lines' => 7,
+			),
+			array(
+				'slug' => 'batcache-variant-error',
+				'level' => BaseScanner::LEVEL_BLOCKER,
+				'description' => 'Variant determiner should refer to at least one $_ variable.',
+				'file' => 'BatcacheVariantTest.inc',
+				'lines' => 7,
+			),
```

Comment:
Is there a better way to check for the error than duplicating the entire error here? What if we change the message in the check itself?